### ISSUE TITLE
Integrate Linguist addon to enable offline pages translation

### ISSUE THREAD

**Issue by vitonsky** [focus-browser/browser#245]
Let's integrate an [Linguist addon](https://github.com/translate-tools/linguist) to enable offline pages translation.
Linguist uses [Bergamot translator](https://github.com/browsermt/bergamot-translator/) (project of mozilla), looks minimalist and have most powerful translation features:
- full page translation
- selected text translation
- user input text translation
- dictionary to save important translations

FOSS and private translation it is important feature for any browser for people. Like powerful and non-commercial ad blocker like uBlock

**Comment by trickypr** [comment]
Thanks for your suggestion. It might be a little while before we can add it, because I will need some time to check over the code. All of the other add-ons included are big enough that someone else has audited (e.g. ublock) or I have a good idea of their internals (e.g. tabliss).

**Comment by vitonsky** [comment]
Of course. If you will have questions, you can add comments here or make issue about audit in a Linguist repository

**Comment by Ryosuke-Asano** [comment]
On Firefox Nightly 115.0a1, Firefox include Translation feature (offline translation)

![image](https://github.com/pulse-browser/browser/assets/73892113/a5c93e7b-9554-4f4e-929a-59be87337fe7)

**Comment by vitonsky** [comment]
It's good news. However, as i know an embedded translator can only translate full page and nothing more.

For me it looks like embedded ad blockers in mobile browsers versus "uBlock Origin" - embedded ad blockers never works correct for me (and i think they are paid by ad platforms to add some ads to a whitelist), so i'm, as most of my friends always install uBlock origin, to ensure ads blocked well.

The same story with Linguist. It's **powerful all-in-one translation tool focused on privacy**. Users even can [add their own custom translators](https://github.com/translate-tools/linguist/blob/master/docs/CustomTranslator.md), and i have plans (and research) to [implement an offline translation of images on page](https://github.com/translate-tools/linguist/issues/310). At the same time, Linguist is not an all-in-one tool for all, it focus to implement only most common features to translation, but never will be a combine that can clean your room unlike some other browser addons.

This is why i still think it is great idea to integrate Linguist to a browser used by geeks.
